include/atomics/senderCadmium.hpp

```diff
@@ -27,148 +27,178 @@ using namespace cadmium;
+            }
+            for(const auto &x : get_messages<typename defs::control_in>(mbs)) {
+                if(state.model_active == false) {
+                    state.total_packet_number = static_cast <int> (x.value);
+                    if (state.total_packet_number > 0) {
+                        state.packet_number = 1;
+                        state.ack = false;
+                        state.sending = true;
+                        state.alt_bit = state.packet_number % 2;  //set initial alt_bit
+                        state.model_active = true;
+                        state.next_internal = PREPARATION_TIME;
                     }
-                  }
-                }
-              }
-              for(const auto &x : get_messages<typename defs::ackIn>(mbs)){
-                if(state.model_active == true) { 
-                  if (state.alt_bit == static_cast < int > (x.value)) {
-                    state.ack = true;
-                    state.sending = false;
-                    state.next_internal = TIME("00:00:00");
-                  }else{
-                    if(state.next_internal != std::numeric_limits<TIME>::infinity()){
-                      state.next_internal = state.next_internal - e;
+                    else {
+                        if (state.next_internal !=
+                            std::numeric_limits<TIME>::infinity()
+                        ) {
+                            state.next_internal = state.next_internal - e;
+                        }
                     }
-                  }
                 }
-              }  
-                           
-            }
-
-            // confluence transition
-            void confluence_transition(TIME e, typename make_message_bags<input_ports>::type mbs) {
-                internal_transition();
-                external_transition(TIME(), std::move(mbs));
             }
-
-            // output function
-            typename make_message_bags<output_ports>::type output() const {
-              typename make_message_bags<output_ports>::type bags;
-              Message_t out;
-              if (state.sending){
-                out.value = state.packetNum * 10 + state.alt_bit;
-                get_messages<typename defs::dataOut>(bags).push_back(out);
-                out.value = state.packetNum;
-                get_messages<typename defs::packetSentOut>(bags).push_back(out);
-              }else{
-                if (state.ack){
-                  out.value = state.alt_bit;
-                  get_messages<typename defs::ackReceivedOut>(bags).push_back(out);
+            for(const auto &x : get_messages<typename defs::acknowledgement_in>(mbs)) {
+                if(state.model_active == true) {
+                    if (state.alt_bit == static_cast<int>(x.value)) {
+                        state.ack = true;
+                        state.sending = false;
+                        state.next_internal = TIME("00:00:00");
+                    } 

```

Comment:
According to coding conventions, in nested loops, open and close brackets should be in the same column.

Reply:
As we discussed on the last Tuesday, these lines are code has no problem. Therefore, please approve the request.